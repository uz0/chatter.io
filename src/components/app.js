import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Router, Route, browserHistory, IndexRedirect } from 'react-router';

import Layout from '@/components/layout';

import SignIn from '@/components/sign-in_container';
import SignUp from '@/components/sign-up_container';
import ForgotPassword from '@/components/forgot-password_container';
import Chat from '@/components/chat_container';

import store from '@/store';

export default class App extends Component {
  render() {
    return <Provider store={store}>
      <Router history={browserHistory}>
        <Route path="/" component={Layout}>
          <IndexRedirect to="chat" />
          <Route component={Chat} path="chat" />
          <Route component={SignIn} path="sign-in" />
          <Route component={SignIn} path="sign-in/:code" />
          <Route component={SignUp} path="sign-up" />
          <Route component={SignUp} path="sign-up/:code" />
          <Route component={ForgotPassword} path="/forgot-password" />
        </Route>
      </Router>
    </Provider>;
  }
}
