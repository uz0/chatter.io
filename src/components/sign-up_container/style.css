.register {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  min-height: 0;
  padding-left: var(--layout-padding);
  padding-right: var(--layout-padding);
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #122033;
}

.register .logo {
  font-family: var(--font-family-title);
  font-size: 30px;
  font-weight: var(--font-weight-bold);
  color: #fff;
}

.register nav {
  display: flex;
  justify-content: space-between;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
}

.register nav a {
  --height: 40px;
  position: relative;
  height: var(--height);
  line-height: var(--height);
  font-size: 16px;
  font-weight: var(--font-weight-ultrabold);
  color: var(--color);
  transition-property: color;
}

.register nav a::before {
  content: "";
  position: absolute;
  left: 0;
  bottom: 0;
  width: 100%;
  height: 1px;
  background-color: #fff;
  opacity: var(--line-opacity);
  transition-property: opacity;
}

.register nav a:nth-of-type(n+2) {
  margin-left: 20px;
}

.register nav a:not(.active_link) {
  --color: #a0a7b0;
  --line-opacity: 0;
}

.register nav a.active_link {
  --color: #fff;
  --line-opacity: 1;
}

.register .form {
  width: 400px;
  margin-top: 34px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width: 700px) {
  .register .form {
    width: 100%;
  }
}

