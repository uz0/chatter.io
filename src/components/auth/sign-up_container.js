import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import { withTranslation } from 'react-i18next';
import Link from '@/components/link';
import Button from '@/components/button';
import Loading from '@/components/loading';
import Validators from '@/components/form/validators';
import Form from '@/components/new-form/form';
import Input from '@/components/input';
import { actions as notificationActions } from '@/components/notification';
import { actions as storeActions } from '@/store';
import { api } from '@';
import { withRouter } from '@/hoc';
import style from './style.css';

class SignUp extends Component {
  state = {
    isLoading: false,
  };

  login = async (data, onError) => {
    try {
      const { user } = await api.login({
        email: data.email.value,
        password: data.password.value,
      });

      this.props.setCurrentUser(user);
      window.localStorage.setItem('authToken', user.auth_token);
      window.localStorage.setItem('currentUser', JSON.stringify(user));

      this.props.showNotification({
        type: 'success',
        text: this.props.t('registered_success'),
      });

      this.props.pushUrl('/chat');
    } catch (error) {
      console.error(error);
      this.setState({ isLoading: false });
      onError(this.props.t(error.code));
    }
  };

  onSubmit = async (data, onError) => {
    if (data.password.value !== data.confirmPassword.value) {
      onError(this.props.t('passwords_not_equal'), ['password', 'confirmPassword']);
      return;
    }

    this.setState({ isLoading: true });

    try {
      await api.register({
        email: data.email.value,
        password: data.password.value,
      });

      this.login(data, onError);
    } catch (error) {
      console.error(error);
      this.setState({ isLoading: false });

      if (error.code === 'invalid_email') {
        onError(this.props.t(error.code), 'email');
        return;
      }

      onError(this.props.t(error.code));
    }
  };

  componentWillMount() {
    if (this.props.currentUser) {
      this.props.pushUrl('/');
      return;
    }
  }

  render() {
    const fields = {
      email: {
        validations: [
          {
            action: Validators.required,
            text: this.props.t('validation_required', { field: this.props.t('email') }),
          },

          {
            action: Validators.email,
            text: this.props.t('validation_correct', { field: this.props.t('email') }),
          },
        ],
      },

      password: {
        validations: [
          {
            action: Validators.required,
            text: this.props.t('validation_required', { field: this.props.t('password') }),
          },

          {
            action: Validators.minLength(6),
            text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
          },

          {
            action: Validators.contains(' '),
            text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
          },
        ],
      },

      confirmPassword: {
        validations: [
          {
            action: Validators.required,
            text: this.props.t('validation_required', { field: this.props.t('password') }),
          },

          {
            action: Validators.minLength(6),
            text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
          },

          {
            action: Validators.contains(' '),
            text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
          },
        ],
      },
    };

    return <div className={style.auth}>
      {this.state.isLoading &&
        <Loading type="line" isShown className={style.loading} />
      }

      <div className={style.header}>
        <h1 className={style.title}>Unichat</h1>
        <Link to="/sign-in" className={style.link}>Log in</Link>
      </div>

      <div className={style.content}>
        <h3 className={style.subtitle}>Get started</h3>

        <Form
          fields={fields}
          onSubmit={this.onSubmit}
          isLoading={this.state.isLoading}
          className={style.form}
        >
          {
            ({ getInputProps, submitProps }) => <Fragment>
              <Input
                {...getInputProps('email')}
                appearance="_none-classic"
                className={style.input}
                placeholder={this.props.t('email')}
              />

              <Input
                {...getInputProps('password')}
                type="password"
                appearance="_none-classic"
                className={style.input}
                placeholder={this.props.t('password')}
              />

              <Input
                {...getInputProps('confirmPassword')}
                type="password"
                appearance="_none-classic"
                className={style.input}
                placeholder={this.props.t('repeat_password')}
              />

              <div className={style.actions}>
                <Button
                  appearance="_basic-primary"
                  text="Sign up"
                  className={style.submit}
                  {...submitProps}
                />
              </div>
            </Fragment>
          }
        </Form>
      </div>

      <p className={style.copyrights}>Â© Unichat (Beta 1.3)</p>
    </div>;
  }
}

export default compose(
  withRouter,
  withTranslation(),

  connect(
    state => ({
      currentUser: state.currentUser,
    }),

    {
      showNotification: notificationActions.showNotification,
      setCurrentUser: storeActions.setCurrentUser,
    },
  ),
)(SignUp);
