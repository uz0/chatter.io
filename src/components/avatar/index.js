import React from 'react';
import classnames from 'classnames/bind';
import style from './style.css';

const cx = classnames.bind(style);

const Avatar = ({ className, letter, photo, isGroup }) => <div
  className={cx('avatar', className, {'_is-group': isGroup})}
  {...photo ? { style: {'--photo': `url(${photo})`} } : {}}
>
  {letter && letter}
</div>;

export default Avatar;
