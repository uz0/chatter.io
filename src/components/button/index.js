import React, { Component } from 'react';
import classnames from 'classnames/bind';
import Icon from '@/components/icon';
import Loading from '@/components/loading';
import style from './style.css';

const cx = classnames.bind(style);

class Button extends Component {
  render = () => <button
    type={this.props.type}
    ref={this.props.ref}
    className={cx('button', this.props.className)}
    appearance={this.props.appearance}
    onClick={this.props.onClick}
    {...this.props.disabled ? {disabled: true} : {}}
    id={this.props.id}
  >
    {this.props.icon && <Icon name={this.props.icon} />}
    {this.props.text && <span>{ this.props.text }</span>}

    {this.props.isLoading &&
      <Loading isShown className={style.loading} />
    }
  </button>;
}

export default Button;
