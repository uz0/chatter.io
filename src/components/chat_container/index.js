import React, { Component } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import classnames from 'classnames/bind';
import { withNamespaces } from 'react-i18next';
import Subscription from '@/components/subscription';
import Button from '@/components/button';
import Icon from '@/components/icon';
import style from './style.css';

const cx = classnames.bind(style);

class Chat extends Component {
  state = {
    navigationActive: 'all',
    isPanelShown: false,
  };

  chooseTabNavigation = tab => () => this.setState({ navigationActive: tab });
  openPanel = () => this.setState({ isPanelShown: true });
  closePanel = () => this.setState({ isPanelShown: false });

  componentWillMount() {
    if (!this.props.currentUser) {
      this.props.router.push('/sign-in');
      return;
    }
  }

  render() {
    return <div className={style.chat}>
      <div className={style.sidebar}>
        <div className={style.header}>
          <h1>Unichat</h1>
          <div className={style.image} style={{ '--bg-image': 'url(/assets/default-user.jpg)' }} />
          <Button appearance="_fab-divider" icon="add-chat" className={style.button} />
        </div>

        <div className={style.search}>
          <div className={style.search_container}>
            <Icon name="search" />
            <input type="text" placeholder={this.props.t('search')} />
          </div>
        </div>

        <div className={style.navigation}>
          <button
            className={cx({'_is-active': this.state.navigationActive === 'all'})}
            onClick={this.chooseTabNavigation('all')}
          >All</button>

          <button
            className={cx({'_is-active': this.state.navigationActive === 'personal'})}
            onClick={this.chooseTabNavigation('personal')}
          >Personal</button>

          <button
            className={cx({'_is-active': this.state.navigationActive === 'work'})}
            onClick={this.chooseTabNavigation('work')}
          >Work</button>
        </div>

        <div className={style.list}>
          <Subscription className={style.subscription} />
          <Subscription className={style.subscription} />
        </div>
      </div>

      <div className={style.content}>
        <div className={style.header}>
          <button onClick={this.openPanel}>Revolution Product</button>
          <p className={style.count}>12 people</p>
        </div>

        <div className={style.messages} />
        <div className={style.input_container} />
      </div>

      <div className={cx('panel', { '_is-shown': this.state.isPanelShown })}>
        <Button appearance="_icon-transparent" icon="arrow-left" onClick={this.closePanel} className={style.close} />
      </div>
    </div>;
  }
}

export default compose(
  withNamespaces('translation'),

  connect(
    state => ({
      currentUser: state.currentUser,
    }),
  ),
)(Chat);
