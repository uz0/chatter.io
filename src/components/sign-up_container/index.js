import React, { Component } from 'react';
import { connect } from 'react-redux';
import compose from 'recompose/compose';
import { withNamespaces } from 'react-i18next';
import { api } from '@';
import Link from '@/components/link';
import { actions as notificationActions } from '@/components/notification';
import { actions as storeActions } from '@/store';
import { actions as formActions } from '@/components/form';
import Form from '@/components/form/form';
import { withRouter } from '@/hoc';
import Input from '@/components/form/input';
import Validators from '@/components/form/validators';

import style from './style.css';

class SignUp extends Component {
  state = {
    commonError: {},
    isLoading: false,
  };

  login = () => api.login({
    email: this.props.formData.email.value,
    password: this.props.formData.password.value,
  }).then(data => {
    console.log('data', data);
    this.setState({ isLoading: false });
    this.props.formReset('register');
    this.props.setCurrentUser(data.user);
    window.localStorage.setItem('authToken', data.user.auth_token);
    window.localStorage.setItem('currentUser', JSON.stringify(data.user));
    this.props.showNotification(this.props.t('registered_success'));
    this.props.pushUrl('/chat');
  }).catch(error => {
    this.setState({ commonError: this.props.t(error.code), isLoading: false });
    this.props.showNotification(this.props.t(error.code));
  });

  submit = event => {
    event.preventDefault();
    this.setState({ isLoading: true });

    if (this.props.formData.password.value !== this.props.formData.confirmPassword.value) {
      this.setState({ commonError: {text: this.props.t('passwords_not_equal')}, isLoading: false });
      this.props.showNotification(this.props.t('passwords_not_equal'));
      return;
    }

    api.register({
      email: this.props.formData.email.value,
      password: this.props.formData.password.value,
    }).then(() => this.login()).catch(error => {
      this.setState({ commonError: this.props.t(error.code), isLoading: false });
      this.props.showNotification(this.props.t(error.code));
    });
  }

  componentWillMount() {
    if (this.props.currentUser) {
      this.props.router.push('/');
      return;
    }
  }

  render = () => {
    return <div className={style.register}>
      <Link to="/" className={style.logo}>Chatter</Link>

      <nav>
        <Link to="/sign-in">{this.props.t('log_in')}</Link>
        <Link className={style.active_link} to="/sign-up">{this.props.t('sign_up')}</Link>
      </nav>

      <Form
        actions={[
          {text: `${this.props.t('get_started')}`, appearance: '_basic-primary', onClick: this.submit, type: 'submit'},
        ]}

        error={this.state.commonError.text || ''}
        model="register"
        disabled={this.state.isLoading}
        className={style.form}
      >
        <Input
          type="email"
          placeholder={this.props.t('email')}
          model="register.email"
          disabled={this.state.isLoading}
          className={style.input}

          validations={[
            {
              action: Validators.required,
              text: this.props.t('validation_required', { field: this.props.t('email') }),
            },

            {
              action: Validators.email,
              text: this.props.t('validation_correct', { field: this.props.t('email') }),
            },
          ]}
        />

        <Input
          type="password"
          placeholder={this.props.t('password')}
          model="register.password"
          disabled={this.state.isLoading}
          className={style.input}

          validations={[
            {
              action: Validators.required,
              text: this.props.t('validation_correct', { field: this.props.t('password') }),
            },

            {
              action: Validators.minLength(6),
              text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
            },

            {
              action: Validators.contains(' '),
              text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
            },
          ]}
        />

        <Input
          type="password"
          placeholder={this.props.t('confirm_password')}
          model="register.confirmPassword"
          disabled={this.state.isLoading}
          className={style.input}

          validations={[
            {
              action: Validators.required,
              text: this.props.t('validation_correct', { field: this.props.t('password') }),
            },

            {
              action: Validators.minLength(6),
              text: this.props.t('validation_min_length', { field: this.props.t('password'), count: 6 }),
            },

            {
              action: Validators.contains(' '),
              text: this.props.t('validation_contains_spaces', { field: this.props.t('password') }),
            },
          ]}
        />
      </Form>
    </div>;
  }
}

export default compose(
  withRouter,
  withNamespaces('translation'),

  connect(
    state => ({
      currentUser: state.currentUser,
      formData: state.forms.register,
    }),

    {
      formReset: formActions.formReset,
      showNotification: notificationActions.showNotification,
      setCurrentUser: storeActions.setCurrentUser,
    },
  ),
)(SignUp);
