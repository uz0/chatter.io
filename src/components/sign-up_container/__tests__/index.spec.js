import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from '@/components/sign-up_container';

describe('SignUp component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SignUp
      t={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
