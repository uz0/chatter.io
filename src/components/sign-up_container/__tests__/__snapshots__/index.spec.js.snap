// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(CustomLink)
          className="logo"
          to="/"
        >
          Unichat
        </withRouter(CustomLink)>,
        <nav>
          <withRouter(CustomLink)
            to="/sign-in"
          />
          <withRouter(CustomLink)
            className="active_link"
            to="/sign-up"
          />
        </nav>,
        <Connect(Form)
          actions={
            Array [
              Object {
                "appearance": "_basic-primary",
                "onClick": [Function],
                "text": "undefined",
                "type": "submit",
              },
            ]
          }
          className="form"
          disabled={false}
          error=""
          model="register"
        >
          <Connect(Input)
            className="input"
            disabled={false}
            model="register.email"
            type="email"
            validations={
              Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ]
            }
          />
          <Connect(Input)
            className="input"
            disabled={false}
            model="register.password"
            type="password"
            validations={
              Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ]
            }
          />
          <Connect(Input)
            className="input"
            disabled={false}
            model="register.confirmPassword"
            type="password"
            validations={
              Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ]
            }
          />
        </Connect(Form)>,
      ],
      "className": "register",
      "data-88a13161": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Unichat",
          "className": "logo",
          "to": "/",
        },
        "ref": null,
        "rendered": "Unichat",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(CustomLink)
              to="/sign-in"
            />,
            <withRouter(CustomLink)
              className="active_link"
              to="/sign-up"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "to": "/sign-in",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "active_link",
              "to": "/sign-up",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            Object {
              "appearance": "_basic-primary",
              "onClick": [Function],
              "text": "undefined",
              "type": "submit",
            },
          ],
          "children": Array [
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.email"
              type="email"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />,
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.password"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />,
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.confirmPassword"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />,
          ],
          "className": "form",
          "disabled": false,
          "error": "",
          "model": "register",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "input",
              "disabled": false,
              "model": "register.email",
              "placeholder": undefined,
              "type": "email",
              "validations": Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "input",
              "disabled": false,
              "model": "register.password",
              "placeholder": undefined,
              "type": "password",
              "validations": Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "input",
              "disabled": false,
              "model": "register.confirmPassword",
              "placeholder": undefined,
              "type": "password",
              "validations": Array [
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
                Object {
                  "action": [Function],
                  "text": undefined,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(CustomLink)
            className="logo"
            to="/"
          >
            Unichat
          </withRouter(CustomLink)>,
          <nav>
            <withRouter(CustomLink)
              to="/sign-in"
            />
            <withRouter(CustomLink)
              className="active_link"
              to="/sign-up"
            />
          </nav>,
          <Connect(Form)
            actions={
              Array [
                Object {
                  "appearance": "_basic-primary",
                  "onClick": [Function],
                  "text": "undefined",
                  "type": "submit",
                },
              ]
            }
            className="form"
            disabled={false}
            error=""
            model="register"
          >
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.email"
              type="email"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.password"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />
            <Connect(Input)
              className="input"
              disabled={false}
              model="register.confirmPassword"
              type="password"
              validations={
                Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ]
              }
            />
          </Connect(Form)>,
        ],
        "className": "register",
        "data-88a13161": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Unichat",
            "className": "logo",
            "to": "/",
          },
          "ref": null,
          "rendered": "Unichat",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(CustomLink)
                to="/sign-in"
              />,
              <withRouter(CustomLink)
                className="active_link"
                to="/sign-up"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "to": "/sign-in",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "className": "active_link",
                "to": "/sign-up",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actions": Array [
              Object {
                "appearance": "_basic-primary",
                "onClick": [Function],
                "text": "undefined",
                "type": "submit",
              },
            ],
            "children": Array [
              <Connect(Input)
                className="input"
                disabled={false}
                model="register.email"
                type="email"
                validations={
                  Array [
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                  ]
                }
              />,
              <Connect(Input)
                className="input"
                disabled={false}
                model="register.password"
                type="password"
                validations={
                  Array [
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                  ]
                }
              />,
              <Connect(Input)
                className="input"
                disabled={false}
                model="register.confirmPassword"
                type="password"
                validations={
                  Array [
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                    Object {
                      "action": [Function],
                      "text": undefined,
                    },
                  ]
                }
              />,
            ],
            "className": "form",
            "disabled": false,
            "error": "",
            "model": "register",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "disabled": false,
                "model": "register.email",
                "placeholder": undefined,
                "type": "email",
                "validations": Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "disabled": false,
                "model": "register.password",
                "placeholder": undefined,
                "type": "password",
                "validations": Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "disabled": false,
                "model": "register.confirmPassword",
                "placeholder": undefined,
                "type": "password",
                "validations": Array [
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                  Object {
                    "action": [Function],
                    "text": undefined,
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
